File: Movie_Search_App_WebApi/Models/MovieDetails.cs
namespace MovieSearchApp_WebApi.Models
{
    public class MovieDetails : BaseMovie
    {
        public string Plot { get; set; }
        public string ImdbRating { get; set; }

        public MovieDetails(string plot, string imdbRating, string title, string year, string imdbID, string type, string poster)
            : base(title, year, imdbID, type, poster)
        {
            Plot = plot;
            ImdbRating = imdbRating;
        }

        public MovieDetails() : base()
        {
            Plot = string.Empty;
            ImdbRating = string.Empty;
        }
    }
}

File: movie_search_app_react/src/components/MovieDetails.js
import React from 'react';
import '../styles/buttons.css';

const MovieDetails = ({ title, year, plot, imdbRating, poster, onHide, className }) => {
  return (
    <div className={`movie-details ${className}`}>
      <button className="button hide-button" onClick={onHide}>Hide</button>
      <h2>{title} ({year})</h2>
      <img src={poster} alt={`${title} Poster`} />
      <p><strong>IMDB Rating:</strong> {imdbRating}</p>
      <p><strong>Description:</strong> {plot}</p>
    </div>
  );
};

export default MovieDetails;
